darts/utils/timeseries_generation.py

```diff
@@ -278,15 +348,26 @@ def datetime_attribute_timeseries(time_index: Union[pd.DatetimeIndex, TimeSeries
     }
 
     values = getattr(time_index, attribute)
-    if one_hot:
+
+    if one_hot or cyclic:
         raise_if_not(attribute in num_values_dict, "Given datetime attribute not supported"
-                                                   " with one-hot encoding.", logger)
+                                                   " with one-hot or cyclical encoding.", logger)
+
+    if one_hot:
         values_df = pd.get_dummies(values)
         # fill missing columns (in case not all values appear in time_index)
         for i in range(1, num_values_dict[attribute] + 1):
             if not (i in values_df.columns):
                 values_df[i] = 0
         values_df = values_df[range(1, num_values_dict[attribute] + 1)]
+    elif cyclic:
```

Comment:
I think we might have to give the user a warning if they call the function with `attribute='day'` and `cyclic=True` because the period will be 31 days even though not every month is that long. For all the other attributes that's not an issue.